### PR TITLE
github action enforce pr continuous integration rules

### PR DESCRIPTION
**Pull request by FabienArcellier**
implement #233

Github action plays the continuous integration pipeline, i.e. mypy, pytest and npm build when opening a Pull Request.

Peek 2024-02-10 19-50

As a developper, I can it in local using the command ``alfred ci``. I can also run specific part of pipeline using `alfred.ci.mypy` or `alfred ci.pytest`

Peek 2024-02-10 19-55

### More info

the PR also fix the type for windows developper when they use mypy.

### Todo

* [x] configure github action for controlling branches and pull-requests on ubuntu (python 3.10, 3.11, 3.12)
* [x] configure github action for controlling branches and pull-requests on macos (python 3.10, 3.11, 3.12)
* [x] configure github action for controlling branches and pull-requests on windows (python 3.10, 3.11, 3.12)
* [x] implement the ``alfred ci.mypy`` command
* [x] implement the ``alfred ci.pytest`` command
* [x] implement the ``alfred build`` pipeline
* [x] implement the ``alfred ci`` pipeline
* [x] github action badges for continuous integration appear on the readme

### CODE REVIEW

File: .github/workflows/ci-windows.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: ci-windows
+
+on: [ push, pull_request ]
+
+jobs:
+  build:
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        python-version: [ "3.10", "3.11", "3.12" ]
```
Comment (ramedina86) [inline .github/workflows/ci-windows.yml:10]: Do you think it's worth / possible to add 3.9.2? That's the earliest version we're supporting

**Comment by ramedina86** [comment]
Should we coordinate the merge of this with the rest of the dev team?

Since I think it might disturb their work a bit if they're not aware it's going on

File: .github/workflows/ci-windows.yml (hunk 1 above)
Comment (FabienArcellier) [reply .github/workflows/ci-windows.yml:10]: Yes, of course. I was convinced we start version 3.10

**Comment by FabienArcellier** [comment]
> Should we coordinate the merge of this with the rest of the dev team?
> 
> Since I think it might disturb their work a bit if they're not aware it's going on

We can, currently the build is running smoothly : https://github.com/FabienArcellier/streamsync/pull/23

**Comment by FabienArcellier** [comment]
image

**Pull request merged**